Source/objects.h

```diff
@@ -20,44 +21,48 @@ namespace devilution {
 #define MAXOBJECTS 127
 
 struct Object {
-	_object_id _otype;
+	_object_id _otype = OBJ_NULL;
 	Point position;
-	bool _oLight;
-	uint32_t _oAnimFlag;
-	byte *_oAnimData;
-	int _oAnimDelay;      // Tick length of each frame in the current animation
-	int _oAnimCnt;        // Increases by one each game tick, counting how close we are to _pAnimDelay
-	uint32_t _oAnimLen;   // Number of frames in current animation
-	uint32_t _oAnimFrame; // Current frame of animation.
-	uint16_t _oAnimWidth;
-	bool _oDelFlag;
-	int8_t _oBreak;
-	bool _oSolidFlag;
+	bool _oLight = false;
+	uint32_t _oAnimFlag = 0;
+	OptionalClxSpriteList _oAnimData;
+	int _oAnimDelay = 0;      // Tick length of each frame in the current animation
+	int _oAnimCnt = 0;        // Increases by one each game tick, counting how close we are to _pAnimDelay
+	uint32_t _oAnimLen = 0;   // Number of frames in current animation
+	uint32_t _oAnimFrame = 0; // Current frame of animation.
+
+	// TODO: Remove this field, it is unused and always equal to:
+	// (*_oAnimData)[0].width()
+	uint16_t _oAnimWidth = 0;

```

Comment:
Did you intend to do this before marking the PR as ready, or is there a reason it should be done later?

Reply:
It should probably be done in a separate PR where we can also remove a few other fields.
Some extra work will be needed for `HeadlessMode` compatibility, which currently relies on this field as we don't actually load the sprites themselves.